mesatee_services/fns/sgx_trusted_lib/src/trusted_worker/lin_reg.rs

```diff
@@ -0,0 +1,183 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(feature = "mesalock_sgx")]
+use std::prelude::v1::*;
+
+use crate::worker::{FunctionType, Worker, WorkerContext};
+use mesatee_core::{Error, ErrorKind, Result};
+
+use rusty_machine::learning::lin_reg::LinRegressor;
+use rusty_machine::learning::SupModel;
+use rusty_machine::linalg::Matrix;
+use rusty_machine::linalg::Vector;
+
+use serde_derive::Deserialize;
+use serde_json;
+
+#[derive(Deserialize)]
+pub(crate) struct LinRegPayload {
+    input_mode_columns: usize,
+    input_mode_data: String,
+    target_mode_data: String,
+    test_data: String,
+}
+
+pub struct LinRegWorker {
+    worker_id: u32,
+    func_name: String,
+    func_type: FunctionType,
+    input: Option<LinRegInput>,
+}
+
+struct LinRegInput {
+    input_mode_data: Matrix<f64>,
+    target_mode_data: Vector<f64>,
+    test_data: Matrix<f64>,
+}
+
+impl LinRegWorker {
+    pub fn new() -> Self {
+        LinRegWorker {
+            worker_id: 0,
+            func_name: "lin_reg".to_string(),
+            func_type: FunctionType::Single,
+            input: None,
+        }
+    }
+}
+
+impl Worker for LinRegWorker {
+    fn function_name(&self) -> &str {
+        self.func_name.as_str()
+    }
+    fn function_type(&self) -> FunctionType {
+        self.func_type
+    }
+    fn set_id(&mut self, worker_id: u32) {
+        self.worker_id = worker_id;
+    }
+    fn id(&self) -> u32 {
+        self.worker_id
+    }
+    fn prepare_input(
+        &mut self,
+        dynamic_input: Option<String>,
+        _file_ids: Vec<String>,
+    ) -> Result<()> {
+        let payload = dynamic_input.ok_or_else(|| Error::from(ErrorKind::MissingValue))?;
+
+        let lin_reg_payload: LinRegPayload = serde_json::from_str(&payload)
+            .or_else(|_| Err(Error::from(ErrorKind::InvalidInputError)))?;
+
+        let input = parse_input_to_matrix(
+            &lin_reg_payload.input_mode_data,
+            lin_reg_payload.input_mode_columns,
+        )?;
+        let target = data_to_vector(&lin_reg_payload.target_mode_data)?;
+        let test_data = parse_input_to_matrix(
+            &lin_reg_payload.test_data,
+            lin_reg_payload.input_mode_columns,
+        )?;
+        self.input = Some(LinRegInput {
+            input_mode_data: input,
+            target_mode_data: target,
+            test_data: test_data,
+        });
+        Ok(())
+    }
+
+    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+        let input = self
+            .input
+            .take()
+            .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        let mut lin_mod = LinRegressor::default();
+        lin_mod
+            .train(&input.input_mode_data, &input.target_mode_data)
+            .unwrap();
+        let output = lin_mod.predict(&input.test_data).unwrap();
```

Comment:
`unwrap()` could cause runtime panic. Can you remove all unwraps in trusted libs?